### PR TITLE
Task#11 - Feat: Create filter product by prices

### LINKED ISSUE #66: Task#11 - DevTask: Filter Products by prices

### PR DESCRIPTION
**Pull request by TheHienDuong**
- Close #66 
## OVERVIEW
- Create new filter => Filter products by price and convertPriceToNumber
- Update handlePriceFilter to get value for selectedFilter.price
- List products by filterProductsByManufacturerAndPrice 
- Fix wraning mix key id for component
- Update button isSelected  color

## TESTING
- Check out each price filter ( done )
- Check each filter on the manufacturer ( done )
- Combined test of both filters ( done )
- Multi-checker filter manufacturer and filter price ( done )
- Check the results displayed when all filters are removed ( done )

## DEMO
https://github.com/Agility-Internship/Duong-The-Hien-ReactJS/assets/113186169/b403fb56-d54c-4382-b7f6-dbbf7176b7f0

### CODE REVIEW

File: practice-two/src/components/GroupFilterPopover.jsx [hunk 1]
```diff
@@ -23,28 +23,35 @@ import useClickOutsides from '../hook/useClickOutside';
...
 
-  const handleButtonClick = () => {
-    setIsPopoverOpen(!isPopoverOpen);
+  const handleSelectPrice = (e) => {
+    onSelectPrice(e);
   };
 
   return (
-    <div className="relative" ref={popoverRef}>
+    <div key={categories.id} className="relative" ref={popoverRef}>
```
Comment (phattruong-agilityio) [inline practice-two/src/components/GroupFilterPopover.jsx]: why using key here

File: practice-two/src/components/PricesFilterPopover.jsx [hunk 2]
```diff
@@ -19,47 +19,56 @@ import useClickOutside from '../hook/useClickOutside';
...
     setIsPopoverOpen(!isPopoverOpen);
   };
 
+  const handleSelectPrice = (e) => {
+    onSelectPrice(e);
+  };
+
   return (
-    <div className="relative" ref={popoverRef}>
+    <div key={categories.id} className="relative" ref={popoverRef}>
```
Comment (phattruong-agilityio) [inline practice-two/src/components/PricesFilterPopover.jsx]: why using key here

File: practice-two/src/components/GroupFilterPopover.jsx (hunk 1 above)
Comment (TheHienDuong) [reply practice-two/src/components/GroupFilterPopover.jsx]: This is an omission when I fix the error, will be removed

File: practice-two/src/components/PricesFilterPopover.jsx (hunk 2 above)
Comment (TheHienDuong) [reply practice-two/src/components/PricesFilterPopover.jsx]: Yes, removed

**Pull request merged**